.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,116 @@
+        uses: docker/setup-buildx-action@v2
+      
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with: 
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Build and Push
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: 'linux/amd64'
+          push: true
+          tags: nethermindeth/juno:${{ env.DOCKER_IMAGE_TAG }}
+    
+  deploy_to_dev:
+    permissions:
+      id-token: write
+      contents: write
+    needs: [docker_build_and_publish]
+    runs-on: ubuntu-latest
+    environment: 
+      name: Development
+    steps:
+      - name: Repository Dispatch Dev
+        env:
+          EVENT_NAME: juno-dev
+          IMAGE_TAG: ${{ needs.docker_build_and_publish.outputs.IMAGE_TAG }}
+          GOERLI: apps/juno-dev/overlays/dev-goerli-1/config.yaml
+          GOERLI2: apps/juno-dev/overlays/dev-goerli-2/config.yaml
+          INTEGRATION: apps/juno-dev/overlays/dev-integration/config.yaml
+          MAINNET: apps/juno-dev/overlays/dev-mainnet/config.yaml
+        run: |
+          curl -L \
+          -X POST \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: token ${{ secrets.ACCESS_TOKEN }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          https://api.github.com/repos/NethermindEth/argo/dispatches \
+          -d '{"event_type": "${{ env.EVENT_NAME }}", "client_payload":{"name": "${{ env.EVENT_NAME }}", "goerli_config": "${{ env.GOERLI }}", "goerli2_config": "${{ env.GOERLI2 }}", "integration_config": "${{ env.INTEGRATION }}", "mainnet_config": "${{ env.MAINNET }}", "tag": "${{ env.IMAGE_TAG }}"}}'
+
+  deploy_to_staging:
+    needs: [docker_build_and_publish, deploy_to_dev]
+    runs-on: ubuntu-latest
+    environment: 
+      name: Staging
+    steps:
+      - name: Repository Dispatch Staging
+        env:
+          EVENT_NAME: juno-staging
+          IMAGE_TAG: ${{ needs.docker_build_and_publish.outputs.IMAGE_TAG }}
+          GOERLI: apps/juno-staging/overlays/staging-goerli-1/config.yaml
+          GOERLI2: apps/juno-staging/overlays/staging-goerli-2/config.yaml
+          INTEGRATION: apps/juno-staging/overlays/staging-integration/config.yaml
+          MAINNET: apps/juno-staging/overlays/staging-mainnet/config.yaml
+        run: |
+          curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: token ${{ secrets.ACCESS_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/NethermindEth/argo/dispatches \
+            -d '{"event_type": "${{ env.EVENT_NAME }}", "client_payload":{"name": "${{ env.EVENT_NAME }}", "goerli_config": "${{ env.GOERLI }}", "goerli2_config": "${{ env.GOERLI2 }}", "integration_config": "${{ env.INTEGRATION }}", "mainnet_config": "${{ env.MAINNET }}", "tag": "${{ env.IMAGE_TAG }}"}}'
```

Comment:
Can we do not deploy goerli2 anymore? It will be deprecated soon, so there is no need to maintain that anymore.